.github/workflows/main.yml

```diff
@@ -13,27 +13,130 @@ on:
 
 jobs:
   build:
-    runs-on: ${{ matrix.os }}
-
+    name: ${{ matrix.config.name }}
+    runs-on: ${{ matrix.config.os }}
+    defaults:
+      run:
+        shell: ${{ matrix.config.shell }}
     strategy:
+      fail-fast: false
       matrix:
-        compiler: [clang, gcc]
-        os: [ubuntu-latest]
+        config:
+          - name: Linux GCC
+            os: ubuntu-latest
+            compiler: gcc
+            shell: bash
+
+          - name: macOS Clang
+            os: macos-latest
+            compiler: clang
+            shell: bash
+
+          - name: MSYS2 MINGW64
+            os: windows-latest
+            compiler: gcc
+            shell: 'msys2 {0}'
+            msystem: mingw64
+            msys-env: mingw-w64-x86_64
```

Comment:
Is there an opportunity to start using `UCRT64`, which is recommended by msys2?

Reply:
Windows 7 users may need to upgrade to use UCRT: [documentation](https://learn.microsoft.com/en-us/cpp/windows/universal-crt-deployment?view=msvc-170)

Reply:
After some thought, we still have a 32-bit build. Fixed in https://github.com/chocolate-doom/chocolate-doom/pull/1628/commits/a2ff9ada2ee09d17c1616f145112cecd81dd1251